src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -492,5 +504,35 @@ private void OnPreviewGraphCompleted(AsyncTask asyncTask)
         }
        
         #endregion
+
+        public IList<ISerializable> GetOrphanedSerializablesAndClearHistoricalTraceData()
+        {
+            var orphans = new List<ISerializable>();
+
+            if (historicalTraceData == null)
+                return orphans;
+
+            // If a Guid exists in the historical trace data
+            // and there is no corresponding node in the workspace
+            // then add the serializables for that guid to the list of
+            // orphans.
+
+            foreach (var nodeData in historicalTraceData)
+            {
+                var nodeGuid = nodeData.Key;
+
+                if (Nodes.All(n => n.GUID != nodeGuid))
+                {
+                    orphans.AddRange(nodeData.Value.SelectMany(CallSite.GetAllSerializablesFromSingleRunTraceData).ToList());
```

Comment:
This is going to get expensive. Can you add a timer for it, if it turns out ot be too expensive, we'll need to build an index.